test/detection/test_line_counter.py

```diff
@@ -70,3 +73,320 @@ def test_calculate_region_of_interest_limits(
+                [[100, 50, 120, 70], [100, 50, 120, 70]],
+                [[-100, 50, -80, 70], [100, 50, 120, 70]],
+                [[100, 50, 120, 70], [100, 50, 120, 70]],
+            ],
+            [[False, False], [False, False], [True, False]],
+            [[False, False], [True, False], [False, False]],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(-100, 0),
+            ),
+            [
+                [[-50, 70, -40, 50], [-80, -50, -70, -40]],
+                [[-50, -70, -40, -50], [-80, 50, -70, 40]],
+                [[-50, 70, -40, 50], [-80, 50, -70, 40]],
+                [[-50, -70, -40, -50], [-80, 50, -70, 40]],
+                [[-50, 70, -40, 50], [-80, 50, -70, 40]],
+                [[-50, -70, -40, -50], [-80, 50, -70, 40]],
+                [[-50, 70, -40, 50], [-80, 50, -70, 40]],
+                [[-50, -70, -40, -50], [-80, -50, -70, -40]],
+            ],
+            [
+                (False, False),
+                (False, True),
+                (True, False),
+                (False, False),
+                (True, False),
+                (False, False),
+                (True, False),
+                (False, False),
+            ],
+            [
+                (False, False),
+                (True, False),
+                (False, False),
+                (True, False),
+                (False, False),
+                (True, False),
+                (False, False),
+                (True, True),
+            ],
+        ),
+    ],
+)
+def test_line_zone_multiple_detections(
+    vector, bbox_sequence, expected_count_in: List[bool], expected_count_out: List[bool]
+) -> None:
+    line_zone = LineZone(start=vector.start, end=vector.end)
+    for i, bboxes in enumerate(bbox_sequence):
+        detections = mock_detections(
+            xyxy=bboxes,
+            tracker_id=[i for i in range(0, len(bboxes))],
+        )
+        count_in, count_out = line_zone.trigger(detections)
+        assert np.all(count_in == expected_count_in[i])
+        assert np.all(count_out == expected_count_out[i])
+
+
+@pytest.mark.parametrize(
+    "vector, bbox_sequence",
+    [
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, 100),
+            ),
+            [
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+            ],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, 100),
+            ),
+            [
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+            ],
+        ),
+    ],
+)
+def test_line_zone_does_not_count_detections_without_tracker_id(vector, bbox_sequence):
+    line_zone = LineZone(start=vector.start, end=vector.end)
+    for bbox in bbox_sequence:
+        detections = Detections(
+            xyxy=np.array([bbox]).reshape((-1, 4)),
+            tracker_id=np.array([None for _ in range(0, 1)]),
```

Comment:
1. That's just [0]. No need for a for loop.
2. I don't think `None` can appear in `tracker_id`. But let's pretend it can, and I'll track it down in #1092
    * If you've seen `None` inside `tracker_id` when testing on your videos, do tell :slightly_smiling_face:

Reply:
Good idea, fixed.